codegen/module.go

```diff
@@ -269,6 +269,54 @@ func (system *ModuleSystem) ResolveModules(
 					append(resolvedDependencies, dependencyInstance)
 			}
 		}
+
+		// Resolve the class dependencies
+		for _, classInstance := range classInstances {
+			for _, classDependency := range classInstance.Dependencies {
+				moduleClassInstances, ok :=
+					resolvedModules[classDependency.ClassName]
+
+				if !ok {
+					return nil, errors.Errorf(
+						"Invalid class name \"%q\" in dependencies for %s %s",
+						classDependency.ClassName,
+						classInstance.ClassName,
+						classInstance.InstanceName,
+					)
+				}
+
+				// TODO: We don't want to linear scan here
+				var dependencyInstance *ModuleInstance
+
+				for _, instance := range moduleClassInstances {
+					if instance.InstanceName == classDependency.InstanceName {
+						dependencyInstance = instance
+						break
+					}
+				}
+
+				if dependencyInstance == nil {
+					return nil, errors.Errorf(
+						"Unknown %q class depdendency \"%q\""+
```

Comment:
I'm not entirely sure that %q is right at all, but I will try to make it consistent at least.